File: hw/sanchez/u3/hw24AbstractFactory/AbstractFactoryJs/macFactory.js
const GUIFactory = require("./guiFactory");
const MacButton = require("./macButton");
const MacMenu = require("./macMenu");

class MacFactory extends GUIFactory {
    createButton() {
        return new MacButton();
    }

    createMenu() {
        return new MacMenu();
    }
}

module.exports = MacFactory;

File: hw/tipantiza/u3/hw24AbstractFactory/AbstractFactory/src/main/java/ec/edu/espe/abstractfactory/model/MacFactory.java

package ec.edu.espe.abstractfactory.model;

/**
 *
 * @author Alexander Tipantiza, The Softwarrios, @ESPE
 */

public class MacFactory extends GUIFactory {
    public Button createButton() {
        return new MacButton();
    }
    
    public Menu createMenu() {
        return new MacMenu();
    }
}
